src/main/java/org/generationcp/middleware/dao/dms/ExperimentDao.java

```diff
@@ -148,6 +148,24 @@ public ExperimentModel getExperimentByProjectIdAndLocation(final Integer project
 		return null;
 	}
 
+	public ExperimentModel getExperimentByTypeInstanceId(final Integer experimentType, final Integer instanceId) {
+		try {
+			final Criteria criteria = this.getSession().createCriteria(this.getPersistentClass());
+			criteria.add(Restrictions.eq("geoLocation.locationId", instanceId));
+			criteria.add(Restrictions.eq("typeId", experimentType));
+			final List<ExperimentModel> list = criteria.list();
+			if (list != null && !list.isEmpty()) {
+				return list.get(0);
+			}
+		} catch (final HibernateException e) {
+			final String message = "Error at getExperimentByProjectIdAndLocation=" + instanceId
```

Comment:
Modify the error message as: Errot at `getExperimentByTypeInstanceId`

Reply:
Done